### ISSUE TITLE
Bugs with deep algos

### ISSUE THREAD

**Issue by TimotheeMathieu** [rlberry-py/rlberry#111]
I open this issue because there seem to be some instability with deep rl agorithms and I propose we document that and try to resolve it here. Please post your example of a clean code that does not behave as it should here and we will try to solve it.

I begin with DQN:

```python
from rlberry.envs import Chain
from rlberry.envs.benchmarks.ball_exploration import PBall2D
from rlberry.manager import AgentManager
from rlberry.agents.torch import DQNAgent
import numpy as np


env_ctor = PBall2D
env_kwargs = {}

agent = AgentManager(
    DQNAgent,
    (env_ctor, env_kwargs),
    fit_budget=3,
    n_fit=1,
    init_kwargs={"horizon": 3},
    seed=42,
)
agent.fit()
state = env.reset()
agent.agent_handlers[0].policy(state)
```
results in 

```
---------------------------------------------------------------------------

FileNotFoundError                         Traceback (most recent call last)

<ipython-input-2-716414c230d7> in <module>()
     17     seed=42,
     18 )
---> 19 agent.fit()
     20 state = env.reset()
     21 agent.agent_handlers[0].policy(state)


/usr/local/lib/python3.7/dist-packages/torch/serialization.py in __init__(self, name, mode)
    209 class _open_file(_opener):
    210     def __init__(self, name, mode):
--> 211         super(_open_file, self).__init__(open(name, mode))
    212 
    213     def __exit__(self, *args):

FileNotFoundError: [Errno 2] No such file or directory: 'rlberry_data/temp/manager_data/DQN_2022-02-02_14-06-50_f62f4977/agent_handlers/idx_0'
```

I also tried with larger horizon and fit_budget but I have the same error. It seems that my agent is not dumped as it should (indeed, no rlberry_data appeared so the dumping has a problem).

**Comment by TimotheeMathieu** [comment]
Solved in PR #110 , the problem was the custom save/load function that used torch.save and torch.load that did not work with AgentManager.

**Comment by TimotheeMathieu** [comment]
New problem is that DQNAgent is not reproducible (i.e. when I give 2 times the same seed, I don't get 2 times the same actions).
